examples/test_util/ESmry.cpp

```diff
@@ -0,0 +1,436 @@
+                throw std::invalid_argument(message);
+            }
+
+            std::vector<int> ministep = unsmry.get<int>(i);
+
+            i++;
+
+            if (std::get<0>(list1[i]) != "PARAMS") {
+                std::string message="Reading summary file, expecting keyword PARAMS, found '" + std::get<0>(list1[i]) + "'";
+                throw std::invalid_argument(message);
+            }
+
+            std::vector<float> tmpData = unsmry.get<float>(i);
+
+            time = tmpData[0];
+
+            if (time == 0.0) {
+                seqTime.push_back(time);
+                seqIndex.push_back(step);
+            }
+
+            i++;
+
+            if (i < static_cast<int>(list1.size())){
+                if (std::get<0>(list1[i]) == "SEQHDR") {
+                    i++;
+                    reportStepNumber++;
+                    seqTime.push_back(time);
+                    seqIndex.push_back(step);
+                }
+            } else {
+                reportStepNumber++;
+                seqTime.push_back(time);
+                seqIndex.push_back(step);
+            }
+            
+            // adding defaut values (0.0) in case vector not found in this particular summary file
+            
+            for (size_t i = 0; i < param.size(); i++){            
+                param[i].push_back(0.0);
+            }
+
+            for (size_t j = 0; j < tmpData.size(); j++) {
+                int ind = arrayInd[n][j];
+                
+                if (ind > -1) {
+                   param[ind][step] = tmpData[j];        
+                }
+            }
+            
+            if (reportStepNumber >= toReportStepNumber) {
+                i = static_cast<int>(list1.size());
+            }
+
+            step++;
+        }
+
+        fromReportStepNumber = toReportStepNumber;
+
+        n--;
+    }
+    
+    nVect = keywList.size();
+    
+    for (auto keyw : keywList){
+        keyword.push_back(keyw);
+    }
+};
+
+
+void ESmry::getRstString(const std::vector<std::string>& restartArray, std::string& pathRst, std::string& rootN) const {
+
+    rootN = "";
+
+    for (auto str : restartArray) {
+        rootN = rootN + str;
+    }
+
+    updatePathAndRootName(pathRst, rootN);
+}
+
+void ESmry::updatePathAndRootName(std::string& dir, std::string& rootN) const {
+
+    if (rootN.substr(0,2) == "./") {
+        rootN = rootN.substr(2, rootN.size() - 2);
+    }
+
+    if (rootN.substr(0,1) == "/") {
+        int p = rootN.find_last_of("/");
+        dir = rootN.substr(0, p);
+        rootN = rootN.substr(p + 1, rootN.size() - p - 1);
+    } else if (rootN.find_first_of("/") != std::string::npos) {
+        int p = rootN.find_last_of("/");
+        dir = dir + "/" + rootN.substr(0, p);
+        rootN = rootN.substr(p + 1, rootN.size() - p - 1);
+    };
+}
+
+bool ESmry::hasKey(const std::string &key) const
+{
+    auto it=std::find(keyword.begin(), keyword.end(), key);
```

Comment:
I personally prefer to use:
```
return (keyword.count(key) > 0);
```
to check if a key is in a map - that is slightly less verbose than the iterator based solution.

Reply:
keyword is not a map, it's a vector.

Reply:
OK - I would really say that a map was the natural container to use here; but then I understand the code at least.